react-components/src/components/Reveal3DResources/ResourceUpdater.ts

```diff
@@ -0,0 +1,330 @@
+
+  private readonly _image360StylingHandler: Image360StylingHandler;
+
+  private readonly _onModelLoaded: (() => void) | undefined;
+  private readonly _onModelLoadedError:
+    | ((addOptions: AddResourceOptions, error: any) => void)
+    | undefined;
+
+  constructor(
+    viewer: Cognite3DViewer,
+    client: CogniteClient,
+    fdmNodeCache: FdmNodeCache,
+    assetMappingCache: AssetMappingCache,
+    pointCloudAnnotationCache: PointCloudAnnotationCache,
+    onModelLoaded: (() => void) | undefined,
+    onModelLoadedError: ((addOptions: AddResourceOptions, error: any) => void) | undefined
+  ) {
+    this._viewer = viewer;
+    this._client = client;
+    this._fdmNodeCache = fdmNodeCache;
+    this._assetMappingCache = assetMappingCache;
+    this._pointCloudAnnotationCache = pointCloudAnnotationCache;
+    this._onModelLoaded = onModelLoaded;
+    this._onModelLoadedError = onModelLoadedError;
+    this._image360StylingHandler = new Image360StylingHandler(this._viewer);
+  }
+
+  public async updateModels(modelAddOptions: AddResourceOptions[]): Promise<void> {
+    this._runningCounter++;
+    const cachedRunningCounter = this._runningCounter;
+
+    const currentlyAddedResources = await this._pendingModelsPromise;
+
+    const isLatestSyncRun = this._runningCounter === cachedRunningCounter;
+    if (!isLatestSyncRun) {
+      return;
+    }
+
+    if (
+      currentlyAddedResources.length !==
+      this._viewer.models.length + this._viewer.get360ImageCollections().length
+    ) {
+      this.clearAllModels();
+      currentlyAddedResources.splice(0);
+    }
+
+    const { newResources, outdatedResources } = findNewAndOutdatedResources(
+      modelAddOptions,
+      currentlyAddedResources.map((options) => options.addOptions)
+    );
+
+    const keptModelsWithOptions = this.removeResources(outdatedResources, currentlyAddedResources);
+    this.updateKeptModels(keptModelsWithOptions, modelAddOptions);
+    const newModelWithAddOptionsPromises = this.addResources(newResources);
+
+    this._pendingModelsPromise = this.constructNewModelPromises(
+      newModelWithAddOptionsPromises,
+      keptModelsWithOptions
+    );
+  }
+
+  public async updateCommonStyling(
+    instanceStyling: InstanceStylingGroup[],
+    defaultStyling: DefaultResourceStyling
+  ): Promise<void> {
+    this._defaultResourceStyling = defaultStyling;
+    this._instanceStyling = instanceStyling;
+
+    this._image360StylingHandler.setCommonStyling(defaultStyling.image360, instanceStyling);
+
+    const models = await this._pendingModelsPromise;
+    models.forEach(async (model) => {
+      await this.applyStylingAndTransform(model);
+    });
+  }
+
+  private clearAllModels(): void {
+    this._viewer.models.forEach((model) => {
+      this._viewer.removeModel(model);
+    });
+    this._viewer.get360ImageCollections().forEach((collection) => {
+      this._viewer.remove360ImageSet(collection);
+    });
+  }
+
+  private updateKeptModels(
```

Comment:
```suggestion
  private updateExistingModels(
```

Reply:
Turns out I don't need this function anymore (at least not for now), so I removed it altogether